src/client/texture.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+Minetest
+Copyright (C) 2019 kilbith, Jean-Patrick Guerrero <jeanpatrick.guerrero@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "texture.h"
+#include "porting.h"
+
+void Texture::set(const std::string &_name, const std::string &base_name,
+		s32 _frame_count, u64 _frame_duration)
+{
+	m_name = _name;
+	m_frame_count = _frame_count;
+	m_frame_idx = 0;
+	m_frame_duration = _frame_duration;
+	m_frame_time = 0;
+	m_frame_names.resize(m_frame_count);
+
+	for (s32 i = 0; i < m_frame_count; ++i)
+		m_frame_names[i] = base_name + "_" + (char)(i + 1 + '0') + ".png";
```

Comment:
This cast won't work correctly for `m_frame_count` greater than 9. I suggest using something like [`std::to_string` ](https://en.cppreference.com/w/cpp/string/basic_string/to_string) instead:
```suggestion
		m_frame_names[i] = base_name + "_" + std::to_string(i + 1) + ".png";
```

Reply:
Good catch! I've applied the fix and updated the test mod to use 10+ frames to confirm that it works.